### PR TITLE
[improvement](column) Support for the default value of current_timestamp in milliseconds

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 43.67 seconds
 stream load tsv:          454 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          69.8 seconds inserted 10000000 Rows, about 143K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230704105530_clickbench_pr_172032.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.17 seconds
 stream load tsv:          458 seconds loaded 74807831229 Bytes, about 155 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          69.7 seconds inserted 10000000 Rows, about 143K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230704122307_clickbench_pr_172068.html

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.59 seconds
 stream load tsv:          450 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          69.2 seconds inserted 10000000 Rows, about 144K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230705033030_clickbench_pr_172421.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.38 seconds
 stream load tsv:          454 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          68.3 seconds inserted 10000000 Rows, about 146K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230705051121_clickbench_pr_172453.html

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 1]
```diff
@@ -104,14 +111,51 @@ public DefaultValue(boolean isSet, String value, String exprName) {
         // default "value", "[]" means empty array
         public static DefaultValue ARRAY_EMPTY_DEFAULT_VALUE = new DefaultValue(true, "[]");
 
+        public static DefaultValue currentTimeStampDefaultValueWithPrecision(Long precision) {
+            if (precision > 3 || precision < 0) {
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java:115]:
```suggestion
            if (precision > 6 || precision < 0) {
```

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DefaultValueExprDef.java [hunk 2]
```diff
@@ -23,28 +23,44 @@
...
     /**
      * generate a FunctionCallExpr
      * @return FunctionCallExpr of exprName
      */
     public FunctionCallExpr getExpr(Type type) {
-        FunctionCallExpr expr = new FunctionCallExpr(exprName, new FunctionParams(null));
+        List<Expr> exprs = null;
+        if (precision != null) {
+            exprs = Lists.newArrayList();
+            exprs.add(new IntLiteral(3));
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DefaultValueExprDef.java:61]: Why use a constant 3 ?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DefaultValueExprDef.java [hunk 3]
```diff
@@ -23,28 +23,44 @@
...
+
 /**
  * This def used for column which defaultValue is an expression.
  */
 public class DefaultValueExprDef {
     private static final Logger LOG = LogManager.getLogger(DefaultValueExprDef.class);
     @SerializedName("exprName")
     private String exprName;
+    @SerializedName("precision")
+    private Long precision;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DefaultValueExprDef.java:41]: For newly added field, better implement `GsonPostProcessable` and give it a init value

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.57 seconds
 stream load tsv:          500 seconds loaded 74807831229 Bytes, about 142 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          88.5 seconds inserted 10000000 Rows, about 112K ops/s
 storage size: 17162247132 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230706172215_clickbench_pr_173596.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.76 seconds
 stream load tsv:          440 seconds loaded 74807831229 Bytes, about 162 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          56 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          67.0 seconds inserted 10000000 Rows, about 149K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230706155857_clickbench_pr_173888.html

File: regression-test/suites/correctness_p0/test_current_timestamp.groovy [hunk 4]
```diff
@@ -128,4 +128,89 @@ suite("test_current_timestamp") {
 
     qt_stream_load_json5 """ select id, name from ${tableName2} where dt_1 is not NULL order by id; """
     qt_stream_load_json6 """ select count(*) from ${tableName2} where dt_2 is not NULL; """
+
+    def tableName3 = "test_current_timestamp_ms"
+
+    sql """ DROP TABLE IF EXISTS ${tableName3} """
+    sql """
+        CREATE TABLE IF NOT EXISTS ${tableName3}
```
Comment (yiguolei) [inline regression-test/suites/correctness_p0/test_current_timestamp.groovy:136]: please add select * from table and show create table test。

**Comment by zy-kkk** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 51.62 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.7 seconds inserted 10000000 Rows, about 348K ops/s
 storage size: 17168517387 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230710114628_clickbench_pr_175546.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.05 seconds
 stream load tsv:          444 seconds loaded 74807831229 Bytes, about 160 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          25.3 seconds inserted 10000000 Rows, about 395K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230710051828_clickbench_pr_175543.html

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 54.35 seconds
 stream load tsv:          442 seconds loaded 74807831229 Bytes, about 161 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          56 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          25.8 seconds inserted 10000000 Rows, about 387K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230710161953_clickbench_pr_176076.html

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.78 seconds
 stream load tsv:          496 seconds loaded 74807831229 Bytes, about 143 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          28.5 seconds inserted 10000000 Rows, about 350K ops/s
 storage size: 17169349408 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230711004016_clickbench_pr_176080.html

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 5]
```diff
@@ -104,14 +111,60 @@ public DefaultValue(boolean isSet, String value, String exprName) {
         // default "value", "[]" means empty array
         public static DefaultValue ARRAY_EMPTY_DEFAULT_VALUE = new DefaultValue(true, "[]");
 
+        public static DefaultValue currentTimeStampDefaultValueWithPrecision(Long precision) {
+            if (precision > 6 || precision < 0) {
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java:115]:
```suggestion
            if (precision > ScalarType.MAX_DATETIMEV2_SCALE || precision < 0) {
```

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Comment by zy-kkk** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 57.08 seconds
 stream load tsv:          499 seconds loaded 74807831229 Bytes, about 142 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          28.7 seconds inserted 10000000 Rows, about 348K ops/s
 storage size: 17161695635 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230711110647_clickbench_pr_176155.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.81 seconds
 stream load tsv:          443 seconds loaded 74807831229 Bytes, about 161 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          26.2 seconds inserted 10000000 Rows, about 381K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230711034319_clickbench_pr_176184.html

**Comment by zy-kkk** [comment]
run feut

**Pull request merged**